### DISCUSSION TITLE
Graph API: Zoom, Export

### DISCUSSION THREAD

**Discussion by askalione** [f5/unovis#142] [Q&A] [answered]
Hello!

Thank you for awesome library! ❤️

I want to display custom control panel (buttons) for graph, e.g.:
![image](https://user-images.githubusercontent.com/3993988/220148778-c0561d29-b059-44f1-9944-243734ba8210.png)

**1. Zoom In / Zoom Out**
How to zoom in/zoom out used graph object?
Can i call something like `graph.zoom()` for zoom in and zoom out? Which API method i can call?

**2. Export to SVG/PNG images**
How to export rendered graph to SVG or PNG image?

Is it possible?

Unfortunately I can't find examples of this in the documentation.
Thanks for any help.

**Comment by askalione** [comment, marked as answer]
Oh, i see - `graph.component.zoomIn();` and `graph.component.zoomOut();` methods!

But what right way to export SVG with all css (default + custom variables)?

**Reply by rokotyan** [reply]
Thanks @askalione!

The library doesn't support SVG/PNG export out of the box, but you can check out my old [example from 2017](https://gist.github.com/rokotyan/0556f8facbaf344507cdc45dc3622177) that might still work (haven't used that code since then).
